NEWS.md

```diff
@@ -1,3 +1,18 @@
+# Development version (unreleased)
+
+* `[setup-r]`: on R 3.6.x we now use the P3M snapshot from 2024-06-01 by
+  default. This is because many newer CRAN packages do not work on R 3.6.x
+  any more. The default CRAN mirror is still added, so newer packages may
+  be used if they are required, but `setup-r-dependencies@v2` will prefer
+  the binary packages from P3M.
+
+  Note that this only happens if you opt in to use P3M with the
```

Comment:
Separately, should we consider making this the default? It would be a major change, but I think we're confident in the stability of P3M now, and it is only going to be a user experience improvement for most people?

Reply:
Yeah, but a breaking change for others, so this is a tough decision. 

How about giving a warning that suggests using P3M, unless people explicitly turned it off?

Reply:
Why do you think it would be a breaking change? 

Maybe we could start warning now, and then switch the default in a year or so?

Reply:
Because if we change the repos, different dependency versions will be installed, and this'll likely break some workflows.

Reply:
Oh got it.